Mathlib/Data/ENNReal/Operations.lean

```diff
@@ -536,6 +536,50 @@ theorem iInf_add_iInf (h : ∀ i j, ∃ k, f k + g k ≤ f i + g j) : iInf f + i
       le_iInf₂ fun a a' => let ⟨k, h⟩ := h a a'; iInf_le_of_le k h
     _ = iInf f + iInf g := by simp_rw [iInf_add, add_iInf]
 
+lemma add_biInf' {p : ι → Prop} (f : ι → ℝ≥0∞) :
+    a + ⨅ i, ⨅ _ : p i, f i = ⨅ i, ⨅ _ : p i, a + f i := by
+  simp only [iInf_subtype', add_iInf]
+
+lemma biInf_add' {p : ι → Prop} (f : ι → ℝ≥0∞) :
+    (⨅ i, ⨅ _ : p i, f i) + a = ⨅ i, ⨅ _ : p i, f i + a := by
+  simp only [add_comm, add_biInf']
+
+lemma add_biInf {ι : Type*} {s : Set ι} (f : ι → ℝ≥0∞) :
+    a + ⨅ i ∈ s, f i = ⨅ i ∈ s, a + f i := add_biInf' _
+
+lemma biInf_add {ι : Type*} {s : Set ι} (f : ι → ℝ≥0∞) :
+    (⨅ i ∈ s, f i) + a = ⨅ i ∈ s, f i + a := biInf_add' _
+
+lemma add_sInf {s : Set ℝ≥0∞}: a + sInf s = ⨅ b ∈ s, a + b := by
+  rw [sInf_eq_iInf, add_biInf]
+
+variable {κ : Sort*}
+
+lemma le_iInf_add_iInf {g : κ → ℝ≥0∞} (h : ∀ i j, a ≤ f i + g j) :
+    a ≤ iInf f + iInf g := by
+  simp_rw [iInf_add, add_iInf]; exact le_iInf₂ h
+
+lemma le_biInf_add_biInf' {p : ι → Prop} {q : κ → Prop}
+    {g : κ → ℝ≥0∞} (h : ∀ i, p i → ∀ j, q j → a ≤ f i + g j) :
+    a ≤ (⨅ i, ⨅ _ : p i, f i) + ⨅ j, ⨅ _ : q j, g j := by
+  simp_rw [biInf_add', add_biInf']
+  exact le_iInf₂ fun i hi => le_iInf₂ (h i hi)
+
+lemma le_biInf_add_biInf {ι κ : Type*} {s : Set ι} {t : Set κ}
+    {f : ι → ℝ≥0∞} {g : κ → ℝ≥0∞} {a : ℝ≥0∞} (h : ∀ i ∈ s, ∀ j ∈ t, a ≤ f i + g j) :
+    a ≤ (⨅ i ∈ s, f i) + ⨅ j ∈ t, g j := le_biInf_add_biInf' h
+
```

Comment:
You're not adding `iInf` versions of these lemmas, intentional?
```
iSup_add_iSup
iSup_add_iSup_of_monotone
sub_iSup

Reply:
`iInf_add_iInf` is already there. I've just added `iInf_add_iInf_of_monotone`. It seems to me that `sub_iSup` hasn't a natural `iInf` version.